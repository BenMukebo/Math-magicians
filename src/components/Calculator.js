import React, { Component } from 'react';

class Calculator extends Component {
  render() {
    return;
  }
}

export default Calculator;