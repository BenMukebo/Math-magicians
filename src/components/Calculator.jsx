import React, { Component } from 'react';
import calculate from '../logic/calculate';

class Calculator extends Component {
  constructor(props) {
    super(props);
    this.state = {
      total: 0,
      next: '',
      operation: '',
    };
  }

  handleCalculate = (value) => this.setState((obj) => calculate(obj, value.target.textContent))

  render() {
    const { total, next, operation } = this.state;
    return (
      <ul className="items-container">
        <li className="flex-center result">
          <div className="input">
            {`
          ${total || operation || next
              ? `${total || ''} ${operation || ''} ${next || ''}`
              : '0'}
            `}
          </div>
        </li>
        <li className="flex-center keyItem">
          <button className="btn" onClick={this.handleCalculate} type="button">AC</button>
          <button className="btn" onClick={this.handleCalculate} type="button">+/-</button>
          <button className="btn" onClick={this.handleCalculate} type="button">&#x25;</button>
          <button className="btn" onClick={this.handleCalculate} type="button">&#xf7;</button>
        </li>
        <li className="flex-center keyItem">
          <button className="btn" onClick={this.handleCalculate} type="button">7</button>
          <button className="btn" onClick={this.handleCalculate} type="button">8</button>
          <button className="btn" onClick={this.handleCalculate} type="button">9</button>
          <button className="btn" onClick={this.handleCalculate} type="button">x</button>
        </li>
        <li className="flex-center keyItem">
          <button className="btn" onClick={this.handleCalculate} type="button">4</button>
          <button className="btn" onClick={this.handleCalculate} type="button">5</button>
          <button className="btn" onClick={this.handleCalculate} type="button">6</button>
          <button className="btn" onClick={this.handleCalculate} type="button">-</button>
        </li>
        <li className="flex-center keyItem">
          <button className="btn" onClick={this.handleCalculate} type="button">1</button>
          <button className="btn" onClick={this.handleCalculate} type="button">2</button>
          <button className="btn" onClick={this.handleCalculate} type="button">3</button>
          <button className="btn" onClick={this.handleCalculate} type="button">+</button>
        </li>
        <li className="flex-center keyItem">
          <button className="btn" onClick={this.handleCalculate} type="button">0</button>
          <button className="btn" onClick={this.handleCalculate} type="button">.</button>
          <button className="btn" onClick={this.handleCalculate} type="button">=</button>
        </li>
      </ul>
    );
  }
}
export default Calculator;
