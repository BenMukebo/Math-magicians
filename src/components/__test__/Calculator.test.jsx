import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';

import Calculator from '../Calculator';

describe('Calculator page component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should return the sum of two number form UI', () => {
    render(<Calculator />);
    fireEvent.click(screen.getByText('3'));
    fireEvent.click(screen.getByText('x'));
    fireEvent.click(screen.getByText('7'));
    fireEvent.click(screen.getByText('='));
    const result = screen.getByRole("none");
    expect(result.innerHTML).toBe('21');
  });
});
