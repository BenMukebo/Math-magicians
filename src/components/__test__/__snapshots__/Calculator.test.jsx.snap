// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator page component renders correctly 1`] = `
<div
  className=" section flex-center"
>
  <h2
    className="second-title"
  >
    Let's do some Math
    <span>
       !
    </span>
  </h2>
  <section
    className="calculator-section"
  >
    <div
      className="flex-center result"
    >
      <div
        className="input"
        role="none"
      >
        0
        
        
      </div>
    </div>
    <ul
      className="items-container"
    >
      <li
        className="flex-center keyItem"
      >
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          รท
        </button>
      </li>
      <li
        className="flex-center keyItem"
      >
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          x
        </button>
      </li>
      <li
        className="flex-center keyItem"
      >
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
      </li>
      <li
        className="flex-center keyItem"
      >
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
      </li>
      <li
        className="flex-center keyItem"
      >
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          .
        </button>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </li>
    </ul>
  </section>
</div>
`;
